### ISSUE TITLE
Exit code shows success even though it failed

### ISSUE THREAD

**Issue by gnilrets** [sqlfluff/sqlfluff#1042]
I was running sqlfluff and trying to use a dbt profile.yml file that was not in the default location.  I had a typo in where I was pointing DBT_PROFIELS_DIR to, and SQLFluff failed.  However, the exit code was 0, indicating success.  The exit code should be non-zero.

## Expected Behaviour

Non-zero exit code

## Observed Behaviour

```
root@be96f68a3f47:/dbt-runner/dbt#  DBT_PROFLIES_DIR=~/.dbt_sqlfulff sqlfluff lint models
=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/marts/core/dim_customers.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/marts/core/dim_dates.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/marts/core/fct_orders.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/marts/core/intermediate/customer_orders.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/marts/core/intermediate/customer_payments.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/marts/core/intermediate/order_payments.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/staging/stg_customers.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/staging/stg_orders.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

=== [dbt templater] Compiling dbt project...
WARNING    Unable to lint models/staging/stg_payments.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1421, in lint_path
    self.lint_string(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1203, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 876, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 202, in process
    return self._unsafe_process(fname, in_str, config)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 236, in _unsafe_process
    selected = self.dbt_selector_method.search(
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 128, in dbt_selector_method
    self.dbt_manifest, previous_state=None
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 103, in dbt_manifest
    dbt_macros_manifest = load_macro_manifest(self.dbt_config)
  File "/usr/local/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/usr/local/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 58, in dbt_config
    self.dbt_config = DbtRuntimeConfig.from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 231, in from_args
    project, profile = cls.collect_parts(args)
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 209, in collect_parts
    profile = cls._get_rendered_profile(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/runtime.py", line 188, in _get_rendered_profile
    return Profile.render_from_args(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 423, in render_from_args
    return cls.from_raw_profiles(
  File "/usr/local/lib/python3.8/site-packages/dbt/config/profile.py", line 370, in from_raw_profiles
    raise DbtProjectError(
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'default'

root@be96f68a3f47:/dbt-runner/dbt# echo $?
0
```

## Version
```
root@d76e0ce0d939:/dbt-runner/dbt# sqlfluff --version
sqlfluff, version 0.5.3
```

## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = ansi
#templater = jinja
templater = dbt
rules = None
exclude_rules = L009
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore_templated_areas = True
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}dbt_{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 110
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = consistent

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = consistent

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = consistent

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
```

**Comment by barrywhart** [comment]
I tested this today on the latest SQLFluff code in `main` (commit `826798198aa663739ad81ce7b1b3e4192b4d4ab9`), and the exit code is set correctly. Partial output below:
```
  File "/Users/bhart/.pyenv/versions/3.7.9/envs/sqlfluff-3.7.9/lib/python3.7/site-packages/dbt/config/profile.py", line 389, in from_raw_profiles
    "Could not find profile named '{}'".format(profile_name)
dbt.exceptions.DbtProjectError: Runtime Error
  Could not find profile named 'foobar'
(sqlfluff-3.7.9) ➜  my-dbt echo $?
1
```
